python/meterpreter/meterpreter.py

```diff
@@ -13,6 +13,8 @@
 import threading
 import time
 import traceback
+import requests
```

Comment:
I could be wrong, but I believe the Python Meterpreter payload is meant to have zero dependencies to increase the likelhood that it works in all contexts. i.e. we couldn't have `import requests` here in the payload and expect it to work 👀

Reply:
I see. So I just have to make it an optional feature with try or if statement, so only if socks proxy is configured then it will import request library otherwise it will use UrlLib

Reply:
If we add proxy support, it should really work consistently without any extra dependencies, ever. 
A minimum version of Python maybe, but that shouldn't be necessary here, we've just had to use that in the past for ctypes. We would also need it added to the stagers. As it is right now, this looks like it has the following requirements, none of which are probably obvious to our users.

* The Python Meterpreter is selected
* A HTTP(S) transport is selected (this one is consistent IIRC with the wider proxy support)
* A stageless Meterpreter is selected
* The target has `requests` installed

We should also ensure that the functionality matches the other Meterpreter implementations which IIRC don't support SOCKS, they only support HTTP and I think it's only for HTTP transports.